src/PostgREST/Response.hs

```diff
@@ -0,0 +1,195 @@
+{-# LANGUAGE RecordWildCards #-}
+module PostgREST.Response
+  ( createResponse
+  , deleteResponse
+  , infoResponse
+  , invokeResponse
+  , openApiResponse
+  , readResponse
+  , singleUpsertResponse
+  , updateResponse
+  ) where
+
+import qualified Data.ByteString.Char8     as BS8
+import qualified Data.ByteString.Lazy      as LBS
+import qualified Data.HashMap.Strict       as HashMap
+import qualified Data.Set                  as Set
+import qualified Network.HTTP.Types.Header as HTTP
+import qualified Network.HTTP.Types.Status as HTTP
+import qualified Network.HTTP.Types.URI    as HTTP
+import qualified Network.Wai               as Wai
+
+import qualified PostgREST.OpenAPI          as OpenAPI
+import qualified PostgREST.Query.Statements as Statements
+import qualified PostgREST.RangeQuery       as RangeQuery
+
+import PostgREST.Config                  (AppConfig (..))
+import PostgREST.ContentType             (ContentType (..))
+import PostgREST.DbStructure             (DbStructure (..),
+                                          tablePKCols)
+import PostgREST.DbStructure.Identifiers (QualifiedIdentifier (..),
+                                          Schema)
+import PostgREST.DbStructure.Proc        (ProcDescription)
+import PostgREST.DbStructure.Table       (Table (..))
+import PostgREST.GucHeader               (GucHeader,
+                                          addHeadersIfNotIncluded,
+                                          unwrapGucHeader)
+import PostgREST.Query                   (WriteQueryResult (..))
+import PostgREST.Request.ApiRequest      (ApiRequest (..),
+                                          InvokeMethod (..))
+import PostgREST.Request.Preferences     (PreferCount (..),
+                                          PreferRepresentation (..))
+
+import qualified PostgREST.ContentType as ContentType
+
+import Protolude      hiding (Handler, toS)
+import Protolude.Conv (toS)
+
+
+readResponse :: Bool -> QualifiedIdentifier -> AppConfig  -> ApiRequest -> Statements.ResultsWithCount -> Maybe Int64 -> [GucHeader] -> Maybe HTTP.Status -> Wai.Response
+readResponse headersOnly identifier AppConfig{..} apiRequest@ApiRequest{..} (_, queryTotal, _ , body, _, _) total gucHeaders gucStatus =
```

Comment:
```suggestion
readResponse :: Bool -> QualifiedIdentifier -> ApiRequest -> Statements.ResultsWithCount -> Maybe Int64 -> [GucHeader] -> Maybe HTTP.Status -> Wai.Response
readResponse headersOnly identifier apiRequest@ApiRequest{..} (_, queryTotal, _ , body, _, _) total gucHeaders gucStatus =
```

At first glance this is not used. Correct?

Either I missed something entirely.. or GHC is not picking up the unused wildcard matching here. Didn't we have one of the CI's just complain exactly about such a thing?